art/attacks/evasion/graphite/utils.py

```diff
@@ -0,0 +1,627 @@
+    mask: np.ndarray,
+    lbd: float,
+    theta: np.ndarray,
+    return_pert_and_mask: bool = False,
+    clip: bool = True,
+) -> Tuple[np.ndarray, np.ndarray, np.ndarray]:
+    """
+    Combines the image and noise to create a perturbed image.
+    :param x: Input image.
+    :param mask: Mask image.
+    :param lbd: lambda multiplier for the perturbation.
+    :param theta: The perturbation.
+    :param return_pert_and_mask: Whether to return both the perturbation and mask.
+    :param clip: Whether to clip the perturbation.
+    """
+    import cv2  # lgtm [py/repeated-import]
+
+    theta_full = cv2.resize(theta, (x.shape[1], x.shape[0]))
+    if len(theta_full.shape) < 3:
+        theta_full = theta_full[:, :, np.newaxis]
+    comb = x + lbd * theta_full
+
+    mask_full = cv2.resize(mask, (x.shape[1], x.shape[0]))
+    if len(mask_full.shape) < 3:
+        mask_full = mask_full[:, :, np.newaxis]
+    mask_full = np.where(mask_full > 0.5, 1.0, 0.0)
+
+    if clip:
+        comb = np.clip(comb, 0, 1)
+
+    pert = np.where(mask_full > 0.5, comb, 0)
+    return comb, pert, mask_full
+
+
+def get_transformed_images(
+    x: np.ndarray,
+    mask: np.ndarray,
+    xforms: List[Tuple[float, float, float, int, float, float, float, float, float]],
+    lbd: float,
+    theta: np.ndarray,
+    pts: np.ndarray,
+    net_size: Tuple[int, int],
+    clip_min: float,
+    clip_max: float,
+) -> List[np.ndarray]:
+    """
+    Get transformed images.
+    :param x: Input image.
+    :param mask: Mask image.
+    :param xforms: Transformation parameters.
+    :param lbd: lambda multiplier for the perturbation.
+    :param theta: The perturbation.
+    :param pts: A set of points that will set the crop size in the perspective transform.
+    :param net_size: Size of the image for the network.
+    :param clip_min: Minimum value of an example.
+    :param clip_max: Maximum value of an example.
+    :return: List of transformed images.
+    """
+    att, pert, mask = add_noise(x, mask, lbd, theta, True)
+
+    if len(xforms) == 0:
+        return [convert_to_network(att, net_size, clip_min, clip_max)]
+
+    images = []
+    for xform in xforms:
+        angle, dist, gamma, blur, crop_percent, crop_off_x, crop_off_y, obj_width, focal = xform
+        images.append(
+            apply_transformation(
+                x,
+                mask,
+                pert,
+                angle,
+                dist,
+                gamma,
+                blur,
+                crop_percent,
+                crop_off_x,
+                crop_off_y,
+                pts,
+                net_size,
+                obj_width,
+                focal,
+                clip_min,
+                clip_max,
+            )
+        )
+
+    return images
+
+
+def transform_wb(
+    x: "torch.Tensor",
+    x_adv: "torch.Tensor",
+    mask: "torch.Tensor",
+    xform: Tuple[float, float, float, int, float, float, float, float, float],
+    pts: Optional[np.ndarray],
```

Comment:
Should assign it with a default value of None and move to the end of the arg list.